skaha/src/main/java/org/opencadc/skaha/utils/PosixHelper.java

```diff
@@ -17,15 +45,94 @@ public static String uidMapping(final PosixPrincipal posixPrincipal) {
                              posixPrincipal.getUidNumber());
     }
 
-    /**
-     * The name of the secret that contains the complete mapping of UIDs and GIDs.  This is mounted into the launched
-     * containers, and the init-users-groups.sh script uses it to populate the system files (/etc/passwd and
-     * /etc/group).
-     *
-     * @param sessionId The session identifier to mark the secret.
-     * @return String secret name, never null.
-     */
-    public static String getPosixMapperSecretName(String sessionId) {
-        return "posix-mapping-" + sessionId;
+    static Date currentDateUTC() {
+        return Calendar.getInstance(DateUtil.UTC).getTime();
+    }
+
+    static boolean posixMappingSecretRequiresUpdate() throws Exception {
+        final JSONObject jsonObject = CommandExecutioner.getSecretData(PosixHelper.POSIX_MAPPINGS_SECRET_NAME, K8SUtil.getWorkloadNamespace());
+        if (jsonObject.has(PosixHelper.POSIX_MAPPING_SECRET_LAST_MOD_KEY)) {
+            final String lastModDateString = CommonUtils.decodeBase64(jsonObject.getString(PosixHelper.POSIX_MAPPING_SECRET_LAST_MOD_KEY));
+            final Date lastModDate = DateUtil.getDateFormat(DateUtil.ISO8601_DATE_FORMAT_LOCAL, DateUtil.UTC).parse(lastModDateString);
+            final long currentAgeInMilliseconds = PosixHelper.currentDateUTC().getTime() - lastModDate.getTime();
+            LOGGER.debug("current age of " + PosixHelper.POSIX_MAPPINGS_SECRET_NAME + " is " + currentAgeInMilliseconds + " milliseconds");
+            return currentAgeInMilliseconds >= PosixHelper.ONE_MINUTE_MS;
+        } else {
+            return true;
+        }
+    }
+
+    public static void ensurePosixMappingSecret(final PosixMapperClient posixMapperClient) throws Exception {
+        final String posixMappingSecretName = PosixHelper.POSIX_MAPPINGS_SECRET_NAME;
+
+        if (PosixHelper.posixMappingSecretRequiresUpdate()) {
+            final Path holdingDir = Files.createTempDirectory(Path.of(System.getProperty("java.io.tmpdir")), posixMappingSecretName);
+            final File uidMappingFile = new File(holdingDir.toString(), "uidmap.txt");
+            final File gidMappingFile = new File(holdingDir.toString(), "gidmap.txt");
```

Comment:
I think this stuff needs to be synchronized or handled in a different way to avoid concurrent updates?

Reply:
Does Files.createTempDirectory make it unique somehow to this thread, like using a UUID in the dir name?